projects/packages/videopress/src/client/block-editor/blocks/video/components/player/index.native.js

```diff
@@ -99,33 +109,41 @@ export default function Player( { isSelected, attributes } ) {
 	}, [ preview, isPlayerLoaded, isRequestingEmbedPreview, previewCheckAttempts ] );
 
 	const onSandboxMessage = useCallback( message => {
-		if ( message.event === 'videopress_loading_state' && message.state === 'loaded' ) {
-			setIsPlayerLoaded( true );
+		switch ( message.event ) {
+			case 'videopress_ready':
+				setIsPlayerReady( true );
+				break;
+			case 'videopress_loading_state':
+				setIsPlayerLoaded( message?.state === 'loaded' );
+				break;
 		}
 	}, [] );
 
-	const loadingStyle = {};
-	if ( ! isPreviewReady( preview ) ) {
-		loadingStyle.height = 250;
-	}
-
-	const renderEmbed = () => {
-		if ( html ) {
-			return (
-				<SandBox
-					html={ html }
-					onWindowEvents={ { message: onSandboxMessage } }
-					viewportProps="user-scalable=0"
-				/>
-			);
-		}
-		return <Text>{ __( 'Loading', 'jetpack-videopress-pkg' ) }</Text>;
-	};
+	const loadingOverlay = (
+		<View style={ style[ 'videopress-player__overlay' ] } >
+			<View style={ loadingViewStyle } >
+				<Icon icon={ VideoPressIcon } size={ iconStyle?.size } style={ iconStyle } />
+				<Text style={ style[ 'videopress-player__loading-text' ] }>
+					{ __( 'Loading', 'jetpack-videopress-pkg' ) }
+				</Text>
+			</View>
+		</View>
+	);
 
+	// Show the loading overlay when: 
+	// 1. Player is not ready
+	// 2. Player is loaded but preview is not ready
+	const showLoadingOverlay = ! isPlayerReady || ( isPlayerLoaded && ! isPreviewReady );
+	
 	return (
-		<View style={ [ style[ 'videopress-player' ], loadingStyle ] }>
+		<View style={ [ style[ 'videopress-player' ], { aspectRatio } ] }>
 			{ ! isSelected && <View style={ style[ 'videopress-player__overlay' ] } /> }
```

Comment:
This deselected overlay is still needed to prevent the video from playing when selecting the block.